lib/internal/util.js

```diff
@@ -850,10 +851,36 @@ const encodingsMap = { __proto__: null };
 for (let i = 0; i < encodings.length; ++i)
   encodingsMap[encodings[i]] = i;
 
+/**
+ * Reassigns the .name property of a function.
+ * Should be used when function can't be initially defined with desired name
+ * or when desired name should include `#`, `[`, `]`, etc.
+ * @param {string | symbol} name
+ * @param {Function} fn
+ * @param {object} [descriptor]
+ * @returns {Function} the same function, renamed
+ */
+function assignFunctionName(name, fn, descriptor = kEmptyObject) {
+  if (typeof name === 'symbol') {
+    const symbolDescription = SymbolPrototypeGetDescription(name);
+    name = symbolDescription !== undefined ? `[${symbolDescription}]` : '';
```

Comment:
Since the goal is to assign a name to the function, I can't think of a case where we'd pass a symbol without a description except if we made a mistake.
```suggestion
    assert(symbolDescription !== undefined);
    name = `[${symbolDescription}]`;
```

Reply:
Agreed!